libs/forms/core/src/control-config.types.ts

```diff
@@ -1,19 +1,20 @@
 import { Observable } from 'rxjs';
+import { DynControlMatch } from './control-matchers.types';
 import { DynControlFactoryParams, DynControlParams } from './control-params.types';
 import { DynControlTriggers } from './control-validation.types';
 import { DynControlType } from './control.types';
 
-export type DynConfigPrimitive = any; // FIXME omit functions?
+// a given id to a validator/async-validator/error-handler/function
+export type DynConfigId = string;
 
-// a plain/serializable value
+// plain/serializable arguments (no functions)
+export type DynConfigPrimitive = string | boolean | number | RegExp | { [k: string]: DynConfigPrimitive };
 export type DynConfigArgs = DynConfigPrimitive | DynConfigPrimitive[] | null;
 
-// a given id to the functions
-export type DynConfigId = string;
-
-// a collection of ids with arguments to be used
-export type DynConfigCollection = { [id: string]: DynConfigArgs }
-                                | Array<DynConfigId | [DynConfigId, DynConfigArgs]>;
+// handlers provided can be referenced by id or [id with args]
+export type DynConfigProvider = DynConfigId | [DynConfigId, DynConfigArgs];
```

Comment:
"Handlers" supports Validators, AsyncValidators, Matchers, Conditions, and will support ErrorHandlers too.
They are Factories of Functions to be used inside the workflow (`ValidatorFn`, `AsyncValidatorFn`, etc).
The services invoke them with any set of given arguments from the config, like `['minLenght', 3]`.